features/integration/c2c.feature

```diff
@@ -0,0 +1,115 @@
+    And I can retrieve all inner transactions that were called from the atomic transaction with call graph "[{'randElement(string,application)(byte,byte[17])':'appl'},{'randElement(string,application)(byte,byte[17])':'appl'}]".
+
+    ###### ----- app at context index 2: SlotMachine ----- ######
+    When I build an application transaction with the transient account, the current application, suggested params, operation "create", approval-program "programs/slot_machine.teal", clear-program "programs/one.teal.tok", global-bytes 3, global-ints 1, local-bytes 0, local-ints 0, app-args "", foreign-apps "", foreign-assets "", app-accounts "", extra-pages 0
+    And I sign and submit the transaction, saving the txid. If there is an error it is "".
+    And I wait for the transaction to be confirmed.
+    Given I remember the new application ID.
+    # Need to fund SlotMachine because it pays for calling RandomByte:
+    And I fund the current application's address with 100000000 microalgos.
+
+    Given a new AtomicTransactionComposer
+
+    # First spin() -> (result, witness0, witness1, witness2)
+    Given I add the nonce "Thing One"
+    When I create the Method object from method signature "spin(application,application)(byte[3],byte[17],byte[17],byte[17])"
+    * I create a new method arguments array.
+    * I append the encoded arguments "ctxAppIdx:0,ctxAppIdx:1" to the method arguments array.
+    * I add a nonced method call with the transient account, the current application, suggested params, on complete "noop", current transaction signer, current method arguments.
+
+    # Second spin() -> (result, witness0, witness1, witness2)
+    Given I add the nonce "Thing Two"
+    When I create the Method object from method signature "spin(application,application)(byte[3],byte[17],byte[17],byte[17])"
+    * I create a new method arguments array.
+    * I append the encoded arguments "ctxAppIdx:0,ctxAppIdx:1" to the method arguments array.
+    * I add a nonced method call with the transient account, the current application, suggested params, on complete "noop", current transaction signer, current method arguments.
+
+    # Atomic Transaction Execution and Analysis
+    Then I build the transaction group with the composer. If there is an error it is "".
+    And I gather signatures with the composer.
+    And I execute the current transaction group with the composer.
+    And The composer should have a status of "COMMITTED".
+    And The app should have returned ABI types "(byte[3],byte[17],byte[17],byte[17]):(byte[3],byte[17],byte[17],byte[17])".
+    And I can dig into the resulting atomic transaction execution tree with path "0,0,0"
+    And I can dig into the resulting atomic transaction execution tree with path "0,2,0"
+    And I can dig into the resulting atomic transaction execution tree with path "1,2,0"
+    And I dig into the paths "0,0:0,1:0,2" of the resulting atomic transaction tree I see group ids and they are all the same
```

Comment:
Can we merge this step with the three steps above? It seems like this step already checks if it can dig into that path implicitly

Reply:
👍 (removing lines 106-108)